app/src/main/java/net/aiscope/gdd_app/ui/metadata/MetadataActivity.kt

```diff
@@ -51,44 +49,24 @@ class MetadataActivity : AppCompatActivity() , MetadataView, CaptureFlow {
 
         metadata_save_sample.setOnClickListener {
             coroutineScope.launch {
-                presenter.save(selectedSmearType(), selectedSpecies(), selectedStage())
+                presenter.save(
+                    MetadataMapper.getSmearType(metadata_section_smear_type_radio_group.checkedRadioButtonId),
+                    MetadataMapper.getSpecies(baseContext, metadata_species_spinner.selectedItem),
+                    MetadataMapper.getStage(baseContext, metadata_stage_spinner.selectedItem))
             }
         }
     }
 
-    private fun selectedSmearType(): SmearType {
-        return when (metadata_section_smear_type_radio_group.checkedRadioButtonId) {
-            R.id.metadata_blood_smear_thick -> SmearType.THICK
-            R.id.metadata_blood_smear_thin -> SmearType.THIN
-            else -> throw IllegalStateException(
-                "${metadata_section_smear_type_radio_group.checkedRadioButtonId} radio button id is unknown"
-            )
-        }
-    }
-
-    private fun selectedSpecies(): MalariaSpecies {
-        return when (metadata_species_spinner.selectedItem) {
-            getString(R.string.malaria_species_p_falciparum) -> MalariaSpecies.P_FALCIPARUM
-            getString(R.string.malaria_species_p_vivax) -> MalariaSpecies.P_VIVAX
-            getString(R.string.malaria_species_p_ovale) -> MalariaSpecies.P_OVALE
-            getString(R.string.malaria_species_p_malariae) -> MalariaSpecies.P_MALARIAE
-            getString(R.string.malaria_species_p_knowlesi) -> MalariaSpecies.P_KNOWLESI
-            else -> throw IllegalStateException(
-                "${metadata_species_spinner.selectedItem} species is unknown"
-            )
-        }
-    }
-
-    private fun selectedStage(): MalariaStage {
-        return when (metadata_stage_spinner.selectedItem) {
-            getString(R.string.malaria_stage_ring) -> MalariaStage.RING
-            getString(R.string.malaria_stage_trophozoite) -> MalariaStage.TROPHOZOITE
-            getString(R.string.malaria_stage_schizont) -> MalariaStage.SCHIZONT
-            getString(R.string.malaria_stage_gametocyte) -> MalariaStage.GAMETOCYTE
-            else -> throw IllegalStateException(
-                "${metadata_species_spinner.selectedItem} stage is unknown"
-            )
+    private fun selectSpinnerValue(spinner: AbsSpinner, value: String) {
+        var index = -1
+        for (i in 0 until spinner.adapter.count){
+            if (spinner.adapter.getItem(i) == value) {
+                index = 1
```

Comment:
Ha, I was under the impression that something was strange, and I found it. I have made the same mistake so many times myself 😂 
```suggestion
                index = i
```

Reply:
🤦‍♀️Oops, thanks for catching that.